File: backend/schemas/product.py
"""
Product Schemas
"""
from pydantic import BaseModel
from datetime import date
from typing import List, Optional


class ProductResponse(BaseModel):
    """产品响应Schema"""

    id: int
    product_name: str
    total_purchase_amount: float
    total_order_count: int
    avg_unit_price: float
    last_purchase_date: Optional[date]
    percentage: Optional[float] = None  # 占总采购额的百分比

    class Config:
        from_attributes = True


class ProductListResponse(BaseModel):
    """产品列表响应Schema"""

    total: int
    items: List[ProductResponse]

File: frontend/src/services/product.ts
/**
 * Product API Service
 */
import api from './api';

export interface Product {
  id: number;
  product_name: string;
  total_purchase_amount: number;
  total_order_count: number;
  avg_unit_price: number;
  last_purchase_date: string | null;
  created_at: string;
  updated_at: string;
  percentage?: number;
}

export interface ProductListResponse {
  total: number;
  total_amount: number;
  items: Product[];
}

export interface ProductDetailResponse {
  product: Product;
  recent_orders: any[];
  monthly_trend: {
    month: string;
    amount: number;
    count: number;
  }[];
}

// Get product list
export const getProducts = (params?: {
  sort_by?: string;
  sort_order?: string;
  search?: string;
}): Promise<ProductListResponse> => {
  return api.get('/products', { params });
};

// Get product by ID
export const getProduct = (id: number): Promise<ProductDetailResponse> => {
  return api.get(`/products/${id}`);
};

// Get product by name
export const getProductByName = (name: string): Promise<Product> => {
  return api.get(`/products/name/${encodeURIComponent(name)}`);
};
